CMSgov/mint-app
# [EASI-3811] Backend - Move Agency Help Questions

# EASI-3811

## Changes and Description

- GQL
- Store SQL
- Migration
- Go data model updates
- Resolver updates

### Note:
Some of these aspects are already on the base branch to support early unblocking



## How to test this change

- Build containers off main
  - `git checkout main && scripts/dev down && scripts/dev up && git checkout feature/EASI-3705_backend_move_state_agency_help_question`
- Insert test data
  - See below
- Migrate data
  - `scripts/dev db:migrate`
- Assert that data has migrated from plan ops eval and learning to plan general characteristics
  - The following columns should be moved
    - agency_or_state_help
    - agency_or_state_help_other
    - agency_or_state_note


### Test Data Insertion SQL
```
INSERT INTO model_plan (id, model_name, archived, status, created_dts, modified_dts, created_by, modified_by, abbreviation) VALUES ('aaaaaaaa-d88e-46fa-8440-148cf0d9c63a', 'Test Migration Model', false, 'PLAN_DRAFT', '2024-01-31 17:28:42.107944 +00:00', null, '00000001-0001-0001-0001-000000000001', null, null);
INSERT INTO plan_ops_eval_and_learning (id, model_plan_id, agency_or_state_help, agency_or_state_help_other, agency_or_state_help_note, stakeholders, stakeholders_other, stakeholders_note, helpdesk_use, helpdesk_use_note, contractor_support, contractor_support_other, contractor_support_how, contractor_support_note, iddoc_support, iddoc_support_note, technical_contacts_identified, technical_contacts_identified_detail, technical_contacts_identified_note, capture_participant_info, capture_participant_info_note, icd_owner, draft_icd_due_date, icd_note, uat_needs, stc_needs, testing_timelines, testing_note, data_monitoring_file_types, data_monitoring_file_other, data_response_type, data_response_file_frequency, data_full_time_or_incremental, eft_set_up, unsolicited_adjustments_included, data_flow_diagrams_needed, produce_benefit_enhancement_files, file_naming_conventions, data_monitoring_note, benchmark_for_performance, benchmark_for_performance_note, compute_performance_scores, compute_performance_scores_note, risk_adjust_performance, risk_adjust_feedback, risk_adjust_payments, risk_adjust_other, risk_adjust_note, appeal_performance, appeal_feedback, appeal_payments, appeal_other, appeal_note, evaluation_approaches, evaluation_approach_other, evalutaion_approach_note, ccm_involvment, ccm_involvment_other, ccm_involvment_note, data_needed_for_monitoring, data_needed_for_monitoring_other, data_needed_for_monitoring_note, data_to_send_particicipants, data_to_send_particicipants_other, data_to_send_particicipants_note, share_cclf_data, share_cclf_data_note, send_files_between_ccw, send_files_between_ccw_note, app_to_send_files_to_known, app_to_send_files_to_which, app_to_send_files_to_note, use_ccw_for_file_distribiution_to_participants, use_ccw_for_file_distribiution_to_participants_note, develop_new_quality_measures, develop_new_quality_measures_note, quality_performance_impacts_payment, quality_performance_impacts_payment_note, data_sharing_starts, data_sharing_starts_other, data_sharing_frequency_other, data_sharing_starts_note, data_collection_starts, data_collection_starts_other, data_collection_frequency_other, data_collection_frequency_note, quality_reporting_starts, quality_reporting_starts_other, quality_reporting_starts_note, model_learning_systems, model_learning_systems_other, model_learning_systems_note, anticipated_challenges, created_dts, modified_dts, ready_for_review_dts, ready_for_clearance_dts, status, ready_for_clearance_by, ready_for_review_by, created_by, modified_by, quality_performance_impacts_payment_other, data_sharing_frequency_continually, data_collection_frequency_continually, data_sharing_frequency, data_collection_frequency, quality_reporting_frequency_continually, quality_reporting_frequency_other, quality_reporting_frequency) VALUES ('bbbbbbbb-ed6e-41f2-b70c-86b61bbbe6a0', 'aaaaaaaa-d88e-46fa-8440-148cf0d9c63a', '{YES_STATE, YES_AGENCY_IDEAS, OTHER}', 'Some example other note', 'Some example note', null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, '2024-01-31 17:28:42.107944 +00:00', null, null, null, 'READY', null, null, '00000001-0001-0001-0001-000000000001', null, null, null, null, null, null, null, null, null);
INSERT INTO plan_general_characteristics (id, model_plan_id, is_new_model, resembles_existing_model, resembles_existing_model_how, resembles_existing_model_note, has_components_or_tracks, has_components_or_tracks_differ, has_components_or_tracks_note, alternative_payment_model_types, alternative_payment_model_note, key_characteristics, key_characteristics_other, key_characteristics_note, collect_plan_bids, collect_plan_bids_note, manage_part_c_d_enrollment, manage_part_c_d_enrollment_note, plan_contract_updated, plan_contract_updated_note, care_coordination_involved, care_coordination_involved_description, care_coordination_involved_note, additional_services_involved, additional_services_involved_description, additional_services_involved_note, community_partners_involved, community_partners_involved_description, community_partners_involved_note, geographies_targeted, geographies_targeted_types, geographies_targeted_types_other, geographies_targeted_applied_to, geographies_targeted_applied_to_other, geographies_targeted_note, participation_options, participation_options_note, agreement_types, agreement_types_other, multiple_patricipation_agreements_needed, multiple_patricipation_agreements_needed_note, rulemaking_required, rulemaking_required_description, rulemaking_required_note, authority_allowances, authority_allowances_other, authority_allowances_note, waivers_required, waivers_required_types, waivers_required_note, created_dts, modified_dts, ready_for_review_dts, ready_for_clearance_dts, status, ready_for_clearance_by, ready_for_review_by, created_by, modified_by, current_model_plan_id, existing_model_id) VALUES ('cccccccc-4a38-4a25-986e-deacd2cdbb85', 'aaaaaaaa-d88e-46fa-8440-148cf0d9c63a', null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, '2024-01-31 17:28:42.107944 +00:00', null, null, null, 'READY', null, null, '00000001-0001-0001-0001-000000000001', null, null, null);
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
